### ISSUE TITLE
@accordproject/markdown-transform causes error when used in the browser

### ISSUE THREAD

**Issue by DianaLease** [accordproject/markdown-transform#271]
**Describe the bug**
Cannot use `@accordproject/markdown-transform` in the browser. 
When trying to use it, the error `TypeError: fs.readFileSync is not a function` comes from `markdown-pdf`. It looks like it is because of a `require` statement for `easy-pdf-parser` here: https://github.com/accordproject/markdown-transform/blob/4fdfef15f741a597ed5a377d915a32ff07498c62/packages/markdown-pdf/src/PdfTransformer.js#L17
Note that adding `node: { fs: 'empty' }` to the parent app's webpack config does not solve the issue.

**To Reproduce**
1. Import `@accordproject/markdown-transform` into a React app
2. Try to use the transform (ie. `transform(input, 'slate', 'markdown_cicero')`)

**Expected behavior**
No error.

**Screenshots**
![Screen Shot 2020-08-05 at 5 09 28 PM](https://user-images.githubusercontent.com/20543103/89467895-6de9ac80-d744-11ea-87d4-578854a711f0.png)

**Possible Solutions**
- Do not require `easy-pdf-parser` until we are inside the method that uses it.
- Check if we are in the browser before requiring it.
- Use a different library that does not use `fs`.

**Comment by dselman** [comment]
I think we can refactor the transform to use `pdf.js` (which runs in the browser) directly. Basically I think we can extract all the text content using:
https://github.com/mozilla/pdf.js/blob/95e102c07bc257c2120fd7fd9141762b2c813a1c/src/display/api.js#L896

It will give us more control over how text is mapped to headings etc.

**Comment by dselman** [comment]
```
static async getText(pdfUrl) {
    const pdfDoc = await pdfjsLib.getDocument(pdfUrl);
    const maxPages = pdfDoc.numPages;
    const countPromises = [];
    for (let j = 1; j <= maxPages; j += 1) {
      const pagePromise = pdfDoc.getPage(j);
      countPromises.push(pagePromise.then((page) => {
        const textContent = page.getTextContent();
        return textContent.then(lines => (
          lines.items.map((l) => {
            if (l.str.match(/^\s+$/)) return '\n';
            return l.str;
          }).join('').replace(/\n+/g, '\n')
        ));
      }));
    }

    return Promise.all(countPromises).then(texts => (
      texts.join('')
    ));
  }
```

**Comment by jeromesimeon** [comment]
> I think we can refactor the transform to use `pdf.js` (which runs in the browser) directly. Basically I think we can extract all the text content using:
> https://github.com/mozilla/pdf.js/blob/95e102c07bc257c2120fd7fd9141762b2c813a1c/src/display/api.js#L896
> 
> It will give us more control over how text is mapped to headings and where thematic breaks get inserted (to preserve page transitions). We may even be able to do something with embedded images.

If we take that route, what are our dependencies? (i.e., what might break / what should be tested?). Notably, is this code used when doing cut+paste in the editor?

**Comment by jeromesimeon** [comment]
This should be fixed by #308 I believe?

**Comment by jeromesimeon** [comment]
> This should be fixed by #308 I believe?

Not out of the box... `pdf2json` links to `pdfjs` leading to that error:
```
Uncaught TypeError: fs.readFileSync is not a function
    at eval (webpack:///./node_modules/pdf2json/lib/pdf.js?:62)
    at Array.forEach (<anonymous>)
    at Object.eval (webpack:///./node_modules/pdf2json/lib/pdf.js?:62)
    at eval (webpack:///./node_modules/pdf2json/lib/pdf.js?:499)
    at Object../node_modules/pdf2json/lib/pdf.js (client.bundle.js:23037)
    at __webpack_require__ (client.bundle.js:20)
```

There is documentation for using `pdfjs` in the browser here: https://github.com/mozilla/pdf.js/wiki/Setup-pdf.js-in-a-website

But I'm unsure how to set that up properly through `pdf2json`. Maybe needs a webpack guru to tell whether it's possible and how.

**Comment by jeromesimeon** [comment]
Fixed by #452
